homeassistant/components/recorder/__init__.py

```diff
@@ -72,6 +75,16 @@ def validate_db_url(db_url: str) -> Any:
     if (db_url == SQLITE_URL_PREFIX or ":memory:" in db_url) and not ALLOW_IN_MEMORY_DB:
         raise vol.Invalid("In-memory SQLite database is not supported")
 
+    if MYSQLDB_UTF8MB4_CHARSET not in db_url and (
+        db_url.startswith(MYSQLDB_URL_PREFIX)
+        or db_url.startswith(MYSQLDB_PYMYSQL_URL_PREFIX)
+    ):
+        raise vol.Invalid(
```

Comment:
Maybe we should adjust it for them and throw a warning instead so it still starts up and gives them some time to fix it for a few versions?

Reply:
Why would the user need a few versions to add the utf8mb4 option to the URL?

Reply:
We can also pass a connection arg with the default charset we prefer, in case it is not defined explicitly by the user.

Reply:
Updated to just override so we always set `"charset=utf8mb4"`